File: backend/caching/supabase_client.py
# /backend/caching/supabase_client.py

import os
from supabase import create_client, Client


SUPABASE_URL = os.getenv("SUPABASE_URL")
# --- Load SERVICE KEY instead of ANON KEY ---
SUPABASE_SERVICE_KEY = os.getenv("SUPABASE_SERVICE_KEY")
# --- End Change ---

# --- Update Validation Check ---
if not SUPABASE_URL or not SUPABASE_SERVICE_KEY:
    # Make sure SUPABASE_SERVICE_KEY exists in your .env file!
    print("FATAL ERROR: Supabase URL and/or SUPABASE_SERVICE_KEY not found in environment variables.")
    print("Ensure your root .env file exists and contains these variables.")
    exit(1) # Exit if essential keys are missing
# --- End Change ---

# --- Initialize Client with SERVICE KEY ---
try:
    supabase: Client = create_client(SUPABASE_URL, SUPABASE_SERVICE_KEY) # <-- USE SERVICE KEY HERE
    print("Shared Supabase client initialized successfully (using Service Key).")
except Exception as e:
     print(f"FATAL ERROR: Could not initialize Supabase client: {e}")
     supabase = None # Set to None or handle error as needed
     exit(1)

File: backend/server/utils/supabase_client.js
// backend/server/utils/supabase_client.js

import "dotenv/config"; // loads .env into process.env
import { createClient } from "@supabase/supabase-js";

// Load URL and SERVICE KEY from environment variables
const supabaseUrl = process.env.SUPABASE_URL;
const supabaseServiceKey = process.env.SUPABASE_SERVICE_KEY;

// Debug logging
console.log(`DEBUG supabase_client.js: SUPABASE_URL=${supabaseUrl}`);
console.log(
  `DEBUG supabase_client.js: SUPABASE_SERVICE_KEY loaded? ${!!supabaseServiceKey}`
);

let supabase;

if (!supabaseUrl || !supabaseServiceKey) {
  if (process.env.NODE_ENV === "test") {
    console.warn("⚠️  Skipping Supabase client init in test environment.");
    // Minimal stub so imports don’t break
    supabase = {
      from: () => ({
        select: () => ({ data: null, error: null }),
      }),
      // add any other no-op methods you need for your tests…
    };
  } else {
    console.error(
      "FATAL ERROR: SUPABASE_URL and SUPABASE_SERVICE_KEY are required."
    );
    console.error(
      "Ensure your .env file is loaded and contains both variables."
    );
    process.exit(1);
  }
} else {
  // both vars present → real client
  supabase = createClient(supabaseUrl, supabaseServiceKey);
  console.log("Supabase client initialized with Service Role Key.");
}

export default supabase;
